src/Administration/Resources/app/administration/src/module/sw-cms/component/sw-cms-inherit-wrapper/index.ts

```diff
@@ -0,0 +1,144 @@
+ * @prop {String} field - The specific configuration field within the element to manage inheritance for.
+ * @prop {String} [fieldPath='value'] - The path within the configuration field to bind to, defaults to 'value'.
+ * @prop {String} [label] - An optional label for the input field. Prefer this over the child component's label.
+ */
+export default Shopware.Component.wrapComponentConfig({
+    template,
+    emits: [
+        EVENTS.UPDATE,
+        EVENTS.RESTORE,
+        EVENTS.REMOVE,
+    ],
+    props: {
+        element: {
+            type: Object as PropType<RuntimeSlot & {
+                translated: {
+                    config?: CmsSlotConfig;
+                }
+            }>,
+            required: true,
+        },
+        contentEntity: {
+            type: Object as PropType<{
+                slotConfig?: {
+                    [slotId: string]: CmsSlotConfig;
+                };
+            }>,
+            required: false,
+            default: null,
+        },
+        field: {
+            type: String,
+            required: true,
+        },
+        fieldPath: {
+            type: String,
+            default() {
+                return 'value';
+            },
+        },
+        label: {
+            type: String,
+            required: false,
+        },
+    },
+    data() {
+        return {
+            showModal: false,
+        };
+    },
+    computed: {
+        currentValue: {
+            get() {
+                return get(this.runtimeConfig, this.fullPath);
+            },
+            set(value: string) {
+                set(this.runtimeConfig, this.fullPath, value);
+
+                this.$emit(EVENTS.UPDATE, value);
+            },
+        },
+        baseConfig() {
+            return this.element.translated?.config;
+        },
+        childConfig() {
+            return this.contentEntity?.slotConfig?.[this.element.id];
+        },
+        runtimeConfig() {
+            return this.element.config;
+        },
+        supportsInheritance() {
+            return !!this.contentEntity;
+        },
+        canInheritField() {
+            return has(this.baseConfig, this.fullPath);
+        },
+        /**
+         * Fields are inherited if the layout is used on a content page (product, category, landing page)
+         * and the field is not overridden in the <entity>.slot_config
+         */
+        isInherited() {
+            return this.contentEntity && isUndefined(get(this.childConfig, this.fullPath));
+        },
+        fullPath() {
+            return this.field.concat('.', this.fieldPath);
+        },
+    },
+    methods: {
+        onInheritanceRestore() {
+            this.showModal = false;
+
+            set(this.runtimeConfig, this.fullPath, get(this.baseConfig, this.fullPath, null));
+            unset(this.childConfig, this.field);
+
+            if (isEmpty(this.childConfig)) {
+                unset(this.contentEntity, 'slotConfig');
```

Comment:
Is it intended to remove the entire slotConfig of the content entity when only one slot’s configuration is empty?